File: 3918-check-divisibility-by-digit-sum-and-product/check-divisibility-by-digit-sum-and-product.cpp
class Solution {
public:
    bool checkDivisibility(int n) {
        int sum=0;
        int u =n;
        int pro=1;
        while(n>0){
            sum+=(n%10);
            pro*=(n%10);
            n=n/10;
        }
        return u%(sum+pro)==0;
    }
};

File: 3918-check-divisibility-by-digit-sum-and-product/check-divisibility-by-digit-sum-and-product.py
class Solution:
    def checkDivisibility(self, n: int) -> bool:
        pro=1
        sum=0
        u=n
        while n>0 :
            pro*=(n%10)
            sum+=(n%10)
            n=n//10
        return u%(sum+pro)==0    